src/BaseVault.sol

```diff
@@ -332,6 +332,11 @@ abstract contract BaseVault is IVault, ERC20PermitUpgradeable, AccessControlUpgr
         }
         shares = previewWithdraw(assets);
 
+        IHooks hooks_ = hooks();
+        if (address(hooks_) != address(0)) {
+            hooks_.beforeWithdraw(assets, _msgSender());
```

Comment:
since we decided to implement hooks, we should go all the way here and provide full support.

In the case of beforeWithdraw you can pass in:

(caller, receiver, owner, assets, shares) - for the full context to be available. to the hook.